back/spec/services/projects_finder_admin_service_spec.rb

```diff
@@ -0,0 +1,209 @@
+    end
+  end
+
+  describe 'self.filter_project_manager' do
+    let!(:p1) { create(:project) }
+    let!(:p2) { create(:project) }
+    let!(:p3) { create(:project) }
+
+    let!(:user1) { create(:user, roles: [{ 'type' => 'project_moderator', project_id: p1.id }]) }
+    let!(:user2) { create(:user, roles: [{ 'type' => 'project_moderator', project_id: p2.id }]) }
+
+    it 'returns projects managed by specified users' do
+      result = described_class.filter_project_manager(Project.all, { managers: [user1.id] })
+      expect(result.pluck(:id)).to eq([p1.id])
+    end
+
+    it 'returns all projects when no managers specified' do
+      result = described_class.filter_project_manager(Project.all, {})
+      expect(result.pluck(:id)).to match_array([p1.id, p2.id, p3.id])
+    end
+  end
+
+  describe 'self.search' do
+    let!(:p1) { create(:project, title_multiloc: { 'en' => 'Test Project 1' }) }
+    let!(:p2) { create(:project, title_multiloc: { 'en' => 'Another Project' }) }
+    let!(:p3) { create(:project, title_multiloc: { 'en' => 'Test Project 2' }) }
+
+    it 'filters projects by search term' do
+      result = described_class.search(Project.all, { search: 'Test' })
+      expect(result.pluck(:id)).to match_array([p1, p3].pluck(:id))
+    end
+
+    it 'returns all projects when search term is empty' do
+      result = described_class.search(Project.all, {})
+      expect(result.pluck(:id)).to match_array([p1, p2, p3].pluck(:id))
+    end
+  end
+
+  describe 'self.execute' do
+    describe 'sort: recently_viewed' do
+      let!(:user) { create(:user) }
+      let!(:p1) do
+        project = create_project(Time.zone.today - 30.days, Time.zone.today - 5.days)
+        create_session(project, Time.zone.today - 20.days)
+        project
+      end
+      let!(:p2) do
+        project = create_project(Time.zone.today - 30.days, Time.zone.today + 30.days)
+        create_session(project, Time.zone.today - 30.days)
+        create_session(project, Time.zone.today - 10.days)
+        project
+      end
+      let!(:p3) do
+        project = create_project(Time.zone.today - 30.days, Time.zone.today + 30.days)
+        create_session(project, Time.zone.today - 5.days)
+        project
+      end
+      let!(:p4) do
+        project = create_project(Time.zone.today - 30.days, Time.zone.today + 30.days)
+        s4 = create(:session, created_at: Time.zone.today - 4.days, user_id: create(:user).id)
+        create(
+          :pageview,
+          session_id: s4.id,
+          path: '/en/',
+          created_at: Time.zone.today - 4.days
+        )
+        project
+      end
+
+      def create_session(project, created_at)
+        session = create(:session, created_at: created_at, user_id: user.id)
+        create(
+          :pageview,
+          session_id: session.id,
+          path: "/en/admin/projects/#{project.id}/phases/741874e2-994f-4cfc-adf9-9b83e7f62ae0",
```

Comment:
Does this hardcoded phase ID matter?

Reply:
No, it's just to make sure that the regex captures the first id (the `project.id`) and not the following phase one